### PR TITLE
add faq pop up modal at footer

### PR DESCRIPTION
**Pull request by ananyag309**
#40 

image

### CODE REVIEW

**Comment by Mustafiz04** [comment]
idk why the popup doesn't look good. can we have a new section above the footer? Also, do something like initially, it should only show questions, and when click on a question it should show an answer and close when click again, and so on.

File: src/components/Footer.jsx [hunk 1]
```diff
@@ -150,10 +159,49 @@ function Footer() {
...
+                GlobeHoppin is a platform for discovering travel resources and connecting with fellow travelers.
+              </p>
+              <h3 className="font-semibold mt-3">Q2: How can I contact support?</h3>
+              <p className="text-gray-600 dark:text-gray-400">
+                You can contact support through our contact page.
+              </p>
+              <h3 className="font-semibold mt-3">Q3: Where can I find the terms and conditions?</h3>
+              <p className="text-gray-600 dark:text-gray-400">
+                The terms and conditions are available in the footer section of our website.
+              </p>
```
Comment (Mustafiz04) [inline src/components/Footer.jsx:194]: These questions and answers should be combined into a single json/js file, exported from there, imported here, and looped through to display.

**Review by Mustafiz04 — CHANGES_REQUESTED** [review]

File: src/components/Footer.jsx (hunk 1 above)
Comment (ananyag309) [reply src/components/Footer.jsx:194]: okay

**Comment by Mustafiz04** [comment]
This one is for your learning.
Always try to make 1 commit for 1 pr/feature. It is better to track history and work

Follow these steps to squash commits into one
```
git reset --soft $(git merge-base main HEAD)
git add .
git commit -m 'commit message'
git push origin <branch-name> --force-with-lease
```

**Comment by ananyag309** [comment]
image

kind of like this @Mustafiz04

**Comment by ananyag309** [comment]
@Mustafiz04 done 
image

made a json file as well

File: src/components/Footer.jsx [hunk 2]
```diff
@@ -1,8 +1,52 @@
...
+                    openQuestion === index ? 'max-h-40 opacity-100' : 'max-h-0 opacity-0'
+                  }`}
+                >
+                  <p className="text-gray-300 text-xs leading-relaxed">{faq.answer}</p>
+                </div>
+              </div>
+            ))}
+          </div>
+        </div>
+      </section>
```
Comment (Mustafiz04) [inline src/components/Footer.jsx:47]: You can create a new component name <FAQ /> move all faq logic there and then call the component [here](https://github.com/GlobeHoppin/GlobeHoppin/blob/main/src/pages/Homepage.jsx#L182).

File: src/components/Footer.jsx [hunk 3]
```diff
@@ -116,7 +161,7 @@ function Footer() {
                   fill="currentColor"
                   viewBox="0 0 20 17"
                 >
-                  <path d="M20 1.892a8.178 8.178 0 0 1-2.355.635 4.074 4.074 0 0 0 1.8-2.235 8.344 8.344 0 0 1-2.605.98A4.13 4.13 0 0 0 13.85 0a4.068 4.068 0 0 0-4.1 4.038 4 4 0 0 0 .105.919A11.705 11.705 0 0 1 1.4.734a4.006 4.006 0 0 0 1.268 5.392 4.165 4.165 0 0 1-1.859-.5v.05A4.057 4.057 0 0 0 4.1 9.635a4.19 4.19 0 0 1-1.856.07 4.108 4.108 0 0 0 3.831 2.807A8.36 8.36 0 0 1 0 14.184 11.732 11.732 0 0 0 6.291 16 11.502 11.502 0 0 0 17.964 4.5c0-.177 0-.35-.012-.523A8.143 8.143 0 0 0 20 1.892Z" />
+                  <path d="M20 1.892a8.178 8.178 0 0 1-2.355.635 4.074 4.074 0 0 0 1.8-2.235 8.344 8.344 0 0 0-2.605.98A4.13 4.13 0 0 0 13.85 0a4.068 4.068 0 0 0-4.1 4.038 4 4 0 0 0 .105.919A11.705 11.705 0 0 1 1.4.734a4.006 4.006 0 0 0 1.268 5.392 4.165 4.165 0 0 1-1.859-.5v.05A4.057 4.057 0 0 0 4.1 9.635a4.19 4.19 0 0 1-1.856.07 4.108 4.108 0 0 0 3.831 2.807A8.36 8.36 0 0 1 0 14.184 11.732 11.732 0 0 0 6.291 16 11.502 11.502 0 0 0 17.964 4.5c0-.177 0-.35-.012-.523A8.143 8.143 0 0 0 20 1.892Z" />
```
Comment (Mustafiz04) [inline src/components/Footer.jsx:164]: Also, don't change anything which is not related to this issue.

**Review by Mustafiz04 — CHANGES_REQUESTED** [review]
Small changes. Also please squash commits into 1 using steps I shared with you.

**Comment by ananyag309** [comment]
image

i tried to merge that way but error were coming

**Comment by ananyag309** [comment]
@Mustafiz04 done , i will keep your point in mind from next time 1 commit per pr. Sorry for this time

File: src/pages/Homepage.jsx [hunk 4]
```diff
@@ -177,6 +185,41 @@ function Homepage() {
...
+                      }`}
+                    >
+                      <p className="text-gray-300 text-xs leading-relaxed">{faq.answer}</p>
+                    </div>
+                  </div>
+                ))}
+              </div>
+            </div>
+          </section>
+
```
Comment (Mustafiz04) [inline src/pages/Homepage.jsx:222]: Remove this because you already have it in the faq component.

File: src/pages/Homepage.jsx [hunk 5]
```diff
@@ -177,6 +185,41 @@ function Homepage() {
...
         </div>
       </div>
```
Comment (Mustafiz04) [inline src/pages/Homepage.jsx:224]: add `<FAQ />` here

**Review by Mustafiz04 — CHANGES_REQUESTED** [review]

**Comment by Mustafiz04** [comment]
> @Mustafiz04 done , i will keep your point in mind from next time 1 commit per pr. Sorry for this time

I can still see 8 commits but okay for now.

**Comment by ananyag309** [comment]
@Mustafiz04 is it fine now

**Review by Mustafiz04 — APPROVED** [review]
